src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/EgressTests.cs

```diff
@@ -226,6 +267,121 @@ public async Task SharedConcurrencyLimitTest()
+                    Assert.Equal(HttpStatusCode.OK, operationResult.StatusCode);
+                    Assert.True(operationResult.OperationStatus.Status == OperationState.Running);
+
+                    // Cancel the trace operation
+                    HttpStatusCode deleteStatus = await apiClient.CancelEgressOperation(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, deleteStatus);
+
+                    operationResult = await apiClient.GetOperationStatus(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, operationResult.StatusCode);
+                    Assert.Equal(OperationState.Cancelled, operationResult.OperationStatus.Status);
+
+                    await Assert.ThrowsAsync<IOException>(async () => await drainResponseTask);
+
+                    await appRunner.SendCommandAsync(TestAppScenarios.AsyncWait.Commands.Continue);
+                },
+                configureTool: (toolRunner) =>
+                {
+                    toolRunner.WriteKeyPerValueConfiguration(new RootOptions().AddFileSystemEgress(FileProviderName, _tempDirectory.FullName));
+                });
+        }
+
+        [Fact]
+        public async Task HttpEgressStopTest()
+        {
+            using TemporaryDirectory tempDir = new(_outputHelper);
+
+            await ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                DiagnosticPortConnectionMode.Connect,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (appRunner, apiClient) =>
+                {
+                    int processId = await appRunner.ProcessIdTask;
+
+                    using ResponseStreamHolder responseBox = await apiClient.HttpEgressTraceAsync(processId, durationSeconds: -1);
+
+                    Uri operationUri = responseBox.Response.Headers.Location;
+                    Assert.NotNull(operationUri);
+
+                    // Start saving the stream
+                    string traceFile = Path.Combine(tempDir.FullName, "test.nettrace");
+                    using FileStream traceFileWriter = File.OpenWrite(traceFile);
+
+                    using MemoryStream dstStream = new();
+                    Task drainResponseTask = responseBox.Stream.CopyToAsync(traceFileWriter);
+
+                    // Make sure the operation exists
+                    OperationStatusResponse operationResult = await apiClient.GetOperationStatus(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, operationResult.StatusCode);
+                    Assert.True(operationResult.OperationStatus.Status == OperationState.Running);
+
+                    // Stop the trace operation
+                    HttpStatusCode deleteStatus = await apiClient.StopEgressOperation(operationUri);
+                    Assert.Equal(HttpStatusCode.Accepted, deleteStatus);
+
+                    await drainResponseTask;
```

Comment:
If the http response doesn't end for some reason (e.g. stopping mechanism is broken), then this will hang the test. Consider `.WaitAsync(...)` with some reasonable timeout.